src/modules/fundingManager/oracle/FM_PC_ExternalPrice_Redeeming_v1.sol

```diff
@@ -0,0 +1,698 @@
+    // Modifiers
+
+    //--------------------------------------------------------------------------
+    // Initialization Function
+
+    /// @inheritdoc Module_v1
+    function init(
+        IOrchestrator_v1 orchestrator_,
+        Metadata memory metadata_,
+        bytes memory configData_
+    ) external override(Module_v1) initializer {
+        // Initialize parent contracts
+        __Module_init(orchestrator_, metadata_);
+
+        // Decode config data
+        (
+            address oracleAddress_,
+            address issuanceToken_,
+            address acceptedToken_,
+            uint buyFee_,
+            uint sellFee_,
+            uint maxSellFee_,
+            uint maxBuyFee_,
+            bool isDirectOperationsOnly_
+        ) = abi.decode(
+            configData_,
+            (address, address, address, uint, uint, uint, uint, bool)
+        );
+
+        // Set accepted token
+        _token = IERC20(acceptedToken_);
+
+        // Cache token decimals for collateral
+        _collateralTokenDecimals = IERC20Metadata(address(_token)).decimals();
+
+        if (
+            !ERC165Upgradeable(address(oracleAddress_)).supportsInterface(
+                type(IOraclePrice_v1).interfaceId
+            )
+        ) {
+            revert Module__FM_PC_ExternalPrice_Redeeming_InvalidOracleInterface(
+            );
+        }
+        // Set oracle
+        _oracle = IOraclePrice_v1(oracleAddress_);
+
+        // Initialize base functionality (should handle token settings)
+        _setIssuanceToken(issuanceToken_);
+
+        // Set fees (checking max fees)
+        if (buyFee_ > maxBuyFee_) {
+            revert Module__FM_PC_ExternalPrice_Redeeming_FeeExceedsMaximum(
+                buyFee_, maxBuyFee_
+            );
+        }
+        if (sellFee_ > maxSellFee_) {
+            revert Module__FM_PC_ExternalPrice_Redeeming_FeeExceedsMaximum(
+                sellFee_, maxSellFee_
+            );
+        }
+
+        // Set fees
+        _setBuyFee(buyFee_);
+        _setSellFee(sellFee_);
+
+        _setMaxBuyFee(maxBuyFee_);
+        _setMaxSellFee(maxSellFee_);
+
+        // Set direct operations only flag
+        setIsDirectOperationsOnly(isDirectOperationsOnly_);
+    }
+
+    //--------------------------------------------------------------------------
+    // Modifiers
+
+    /// @notice Modifier to check if direct operations are only allowed
+    modifier onlyDirectOperations() {
+        if (_isDirectOperationsOnly) {
+            revert
+                Module__FM_PC_ExternalPrice_Redeeming_ThirdPartyOperationsDisabled();
+        }
+        _;
+    }
+
+    //----------------------------------------------------------
+    // View Functions
+
+    /// @inheritdoc IFundingManager_v1
+    /// @return token_ The token address
+    function token() public view override returns (IERC20) {
+        return _token;
+    }
+
+    /// @notice Calculates and returns the static price for buying the issuance token.
```

Comment:
```suggestion
    /// @inheritdoc IBondingCurveBase_v1
```

Reply:
I closed the other comment, but this is what I meant by inheriting the natspec se

Reply:
oookay, I see now. Thank you for clarifying